system/lib/wasmfs/backend.h

```diff
@@ -43,4 +43,6 @@ class Backend {
 // them will be defined in a header file. This is so that any unused backends
 // are not linked in if they are not called.
 backend_t createMemoryFileBackend();
+
+typedef backend_t (*make_backend_t)(void*);
```

Comment:
Can we avoid duplicating this typedef?

Reply:
Sadly no. We already do the same for `backend_t` for the same reason. Trying to merge them leads to various C++ errors that complain about the types not being identical. Maybe a general refactoring of these headers could fix that, though.

Reply:
yikes! That would be good to fix in a follow-up.